asap/asaplib/cluster/ml_cluster_fit.py

```diff
@@ -101,83 +104,280 @@ def pack(self):
         return self.db.get_params
 
 
+# class LAIO_DB(FitClusterBase):
+#     """Laio Clustering scheme
+#     Clustering by fast search and find of density peaks
+#     https://science.sciencemag.org/content/sci/344/6191/1492.full.pdf
+#
+#     $$ \rho_i\,=\,\sum_j{\chi(d_{ij}-d_{cut})}$$ i.e. the local density of data point x_i
+#     $$ \delta_i\,=\,\min_{j:\rho_j>\rho_i}(d_{ij})$$ i.e. the minimum distance to a neighbour with higher density
+#     """
+#
+#     _pairwise = True
+#
+#     def __init__(self, deltamin=-1, rhomin=-1):
+#         self.deltamin = deltamin
+#         self.rhomin = rhomin
+#
+#     def fit(self, dmatrix, rho=None):
+#
+#         """
+#
+#         Parameters
+#         ----------
+#         dmatrix: The distance matrix of shape (Nele, Nele)
+#         rho: The log densities of the points of shape (Nele,)
+#
+#         Returns
+#         -------
+#
+#         """
+#
+#         if rho is None:
+#             raise ValueError('for fdb it is better to compute kernel density first')
+#
+#         delta, nneigh = self.estimate_delta(dmatrix, rho)
+#
+#         # I'll think about this!!! Criterion for deciding which data points act as cluster centres
```

Comment:
One big difference is that the Density_peaks_clustering class requires the user to specify dens_cut and delta_cut which correspond to self.rhomin and self.deltamin. The best place to resolve this difference isn't clear to me.

Reply:
Any thoughts?

Reply:
Can we add these variables in the __init__ of the clustering base class

Reply:
> Can we keep this old method (Laio_DB) and simply add the new method?

I think it makes the most sense to switch out the entire class. I'm writing a fit method for the new class.



> Can we add these variables in the **init** of the clustering base class

On second thoughts it appears this isn't a huge problem because the variables are computed (in both the old and the new class) using the densities and distances supplied.